### DISCUSSION TITLE
Cannot validate a MultiSelect field

### DISCUSSION THREAD

**Discussion by dominik-eller** [laravel/nova-issues#6250] [Q&A] [answered]
- Laravel Version: 10.46
- Nova Version: 4.27
- PHP Version: 8.3
- Database Driver & Version: MYSQL
- Operating System and Version: MAC OS Sonoma 14.2.1
- Browser type and version: Safari 17.2.1
- Reproduction Repository: -/-

### Description:
I am trying to make a MultiSelect field required and I ended up using something like that:

    MultiSelect::make('reason')->options([
                'Option A' => 'Option A',
                'Option B' => 'Option B',
                'Option C' => 'Option C',
            ])->rules('required', 'array', 'min:1'),

I also tried to use 'in:Option A,Option B,Option C' but it is not working. 

I am casting the reason property to array as suggested in the documentation, but the validation message always returns "The reason field must be an array."

### Detailed steps to reproduce the issue on a fresh Nova installation:

- add a MultiSelect field to a nova resource
- add a cast to array to that property
- add a validation to the field

even just using ->rules('required') is not working for MultiSelect field.

**Comment by crynobone** [comment]
The value send on the request is array as `json` so the correct validation rule should be `json`.

**Reply by dominik-eller** [reply]
Thanks for the reply. Do you have a tested / working example of a required validation for a MultiSelect field? I tried json as well before opening the issue (discussion).

**Comment by dominik-eller** [comment, marked as answer]
Ok I found a solution that works for me

    ->rules('required', 'json', function($attribute, $value, $fail) {
                $options = ['Option A', 'Option B', 'Option C'];
                $valueArray = json_decode($value);
                $containsOption = array_intersect($options, $valueArray);
                if(empty($containsOption)) {
                    $fail('The selected reason should contain at least one allowed option.');
                }
            })
